### PR TITLE
Lambda Code Writer Support

### PR DESCRIPTION
**Pull request by ScottCarda-MS**
Adds support for processing lambdas to the Code Writer transformation.

### CODE REVIEW

File: src/QsCompiler/Transformations/QsharpCodeOutput.cs [hunk 1]
```diff
@@ -528,6 +528,37 @@ private static string ReplaceInterpolatedArgs(string text, Func<int, string> rep
                 return InterpolationArg.Replace(text, ReplaceMatch);
             }
 
+            private static string QsSymbolToString(QsSymbol symbol)
+            {
+                if (symbol.Symbol is QsSymbolKind<QsSymbol>.Symbol sym)
```
Comment (ScottCarda-MS) [inline src/QsCompiler/Transformations/QsharpCodeOutput.cs:533]: I believe I need to add a case here for missing lambda params "`_`".

Comment (ScottCarda-MS) [reply src/QsCompiler/Transformations/QsharpCodeOutput.cs:533]: done

File: src/QsCompiler/Transformations/QsharpCodeOutput.cs [hunk 2]
```diff
@@ -528,6 +528,49 @@ private static string ReplaceInterpolatedArgs(string text, Func<int, string> rep
...
+                        return QsSymbolToString(tup.Item.First());
+                    }
+                    else
+                    {
+                        return $"({string.Join(", ", tup.Item.Select(QsSymbolToString))})";
+                    }
+                }
+                else if (symbol.Symbol.IsOmittedSymbols)
+                {
+                    return "__omittedSymbol__";
```
Comment (bettinaheim) [inline src/QsCompiler/Transformations/QsharpCodeOutput.cs:558]: Omitted symbols are generally three dots, so you can replace that with "..."

File: src/QsCompiler/TextProcessor/QsKeywords.fs [hunk 3]
```diff
@@ -304,6 +304,9 @@ let qsNEGop = QsOperator.New("-", 45, false)
 let qsSetUnion = QsOperator.New("+", 10, true)
 let qsSetIntersection = QsOperator.New("*", 20, true)
 
+let qsLambdaOp = QsOperator.New("=>", 0, false)
```
Comment (bettinaheim) [inline src/QsCompiler/TextProcessor/QsKeywords.fs:307]: Seeing that this has not yet been in it so far, I need to double check the parser, and whether it hardcodes the arrows. The precedence here needs to be examined carefully. Could you please add to the task list for the lambda issue to clarify the (pseudo-)precedence of the lambda operator, and remove it from here until we do that (and hard code it in the code write for now) - I can take a look as part of the PR around the data structure that I have in work.

**Review by bettinaheim — APPROVED** [review]
I'll approve the PR to facilitate testing, but I also expect to largely revise this particular change along with a change to use something else than QsSymbol to represent the argument tuple. Please see also my other comment on this PR.

File: src/QsCompiler/TextProcessor/QsKeywords.fs (hunk 3 above)
Comment (ScottCarda-MS) [reply src/QsCompiler/TextProcessor/QsKeywords.fs]: Okay, I've commented the QsKeywords.fs ones out and added a ToDo. I've added local hardcoded the ones in code writer with another ToDo.

File: src/QsCompiler/Transformations/QsharpCodeOutput.cs (hunk 2 above)
Comment (ScottCarda-MS) [reply src/QsCompiler/Transformations/QsharpCodeOutput.cs]: updated, thanks

**Pull request merged**